control-plane-operator/hostedclusterconfigoperator/controllers/machine/machine.go

```diff
@@ -180,3 +208,51 @@ func serviceHasIPFamily(service *corev1.Service, ipFamilyToFind corev1.IPFamily)
 	}
 	return false
 }
+
+func kubevirtInfraNamespace(hcp *hyperv1.HostedControlPlane) string {
+	if hcp.Spec.Platform.Kubevirt.Credentials != nil {
+		return hcp.Spec.Platform.Kubevirt.Credentials.InfraNamespace
+	}
+	return hcp.Namespace
+}
+
+func (r *reconciler) removeOrphanKubevirtPassthrowEndpointSlices(ctx context.Context, hcp *hyperv1.HostedControlPlane) error {
+	namespace := kubevirtInfraNamespace(hcp)
+	endpointSliceList := discoveryv1.EndpointSliceList{}
+	managedByControlPlaneOperator := client.MatchingLabels{discoveryv1.LabelManagedBy: managedByValue}
+	if err := r.kubevirtInfraClient.List(ctx, &endpointSliceList, client.InNamespace(namespace), managedByControlPlaneOperator); err != nil {
+		return fmt.Errorf("failed finding endpoint slices managed by control plane operator: %w", err)
+	}
```

Comment:
just double checking, this client is using a cache for the "List" operation here, right?

Reply:
We do this 
```golang
kubevirtInfraCache, err := cache.New(opts.KubevirtInfraConfig, cache.Options{
		Scheme: kubevirtScheme,
		Mapper: kubevirtMapper,
		DefaultNamespaces: map[string]cache.Config{
			kubevirtInfraNamespace(hcp): cache.Config{}},
	})
	if err != nil {
		return fmt.Errorf("failed building kubevirt infra cache: %w", err)
	}
	// if kubevirt infra config is not used, it is being set the same as the mgmt config
	kubevirtInfraClient, err := client.New(opts.KubevirtInfraConfig, client.Options{
		Scheme: kubevirtScheme,
		Mapper: kubevirtMapper,
		Cache:  &client.CacheOptions{Reader: kubevirtInfraCache},
		WarningHandler: client.WarningHandlerOptions{
			SuppressWarnings: true,
		},
```

Reply:
based on the Cache settings, i think we're set. Here's what i saw in the controller runtime documentation
```
// If both Options.Cache and Options.Cache.Reader are non-nil,
// the client reads from a local cache. 
```